app/src/main/java/com/example/campusguide/DirectionsActivity.kt

```diff
@@ -63,33 +72,43 @@ class DirectionsActivity : AppCompatActivity() {
             text = endName
         }
 
-        // Hash map containing (travelMode, path) pairs
-        paths = mapOf(
-            "driving" to createPath(startName, endName, "driving"),
-            "walking" to createPath(startName, endName, "walking"),
-            "transit" to createPath(startName, endName, "transit")
+        // Hash map containing (travelMode, path) pairs for the three main paths
+        mainPaths = mapOf(
+            "driving" to createPath(startName, endName, "driving", null),
+            "walking" to createPath(startName, endName, "walking", null),
+            "transit" to createPath(startName, endName, "transit", null)
+        )
+
+        // Hash map containing (title, path) pairs for the optional transit paths
+        extraPaths = mapOf(
+            "RECOMMENDED ROUTE" to mainPaths.getValue("transit"),
```

Comment:
Keys in hashmaps should definitely be in constants!

Reply:
Done